File: sdk/go/neburst/billing.go
package neburst

// GetBalance returns the authenticated user's account balance.
func (c *Client) GetBalance() (*Balance, error) {
	var result Balance
	err := c.doRequest("GET", "/open/v1/billing/balance", nil, nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// ListInvoices returns invoices for the authenticated user, with pagination.
func (c *Client) ListInvoices(opts ...ListOption) (*PaginatedResult[Invoice], error) {
	q := applyListOpts(opts)
	var result PaginatedResult[Invoice]
	err := c.doRequest("GET", "/open/v1/billing/invoices", q, nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// GetInvoice returns a single invoice by its UUID.
func (c *Client) GetInvoice(id string) (*Invoice, error) {
	var result Invoice
	err := c.doRequest("GET", "/open/v1/billing/invoices/"+id, nil, nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

File: sdk/rust/src/billing.rs
use std::collections::BTreeMap;
use crate::client::NeburstClient;
use crate::types::{Balance, Invoice, PaginatedResult, Result};

impl NeburstClient {
    pub async fn get_balance(&self) -> Result<Balance> {
        self.request("GET", "/open/v1/billing/balance", None, None::<&()>).await
    }

    pub async fn list_invoices(&self, page: i32, page_size: i32) -> Result<PaginatedResult<Invoice>> {
        let mut q = BTreeMap::new();
        q.insert("page".into(), page.to_string());
        q.insert("page_size".into(), page_size.to_string());
        self.request("GET", "/open/v1/billing/invoices", Some(&q), None::<&()>).await
    }

    pub async fn get_invoice(&self, id: &str) -> Result<Invoice> {
        self.request("GET", &format!("/open/v1/billing/invoices/{}", id), None, None::<&()>).await
    }
}
